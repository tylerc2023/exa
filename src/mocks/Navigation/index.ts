export const mockNavigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  push: jest.fn(),
  popToTop: jest.fn(),
};
