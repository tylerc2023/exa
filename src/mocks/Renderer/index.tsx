import React from 'react';
import {Provider} from 'react-redux';
import {ReactTestRenderer, create} from 'react-test-renderer';
import {store} from '../../providers/Redux';

export const mockRenderer = (component: JSX.Element): ReactTestRenderer =>
  create(<Provider store={store}>{component}</Provider>);
