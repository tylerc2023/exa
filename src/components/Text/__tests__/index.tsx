import React from 'react';
import {Text} from '..';
import {mockRenderer} from '../../../mocks/Renderer';

it('renders correctly', () => {
  const {tree} = mockRenderer({component: <Text title="hello" />});
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders empty', () => {
  const {tree} = mockRenderer({component: <Text title="" />});
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders h1', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="h1" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders type="h2"  ', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="h2" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders h3', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="h3" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders h4', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="h4" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders h5', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="h5" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders h6', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="h6" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders subtitle1', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="subtitle1" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders subtitle2', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="subtitle2" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders body1', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="body1" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders body2', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="body2" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders button', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="button" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders button empty', () => {
  const {tree} = mockRenderer({component: <Text title="" type="button" />});
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders caption', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="caption" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
it('renders overline', () => {
  const {tree} = mockRenderer({
    component: <Text title="hello" type="overline" />,
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
