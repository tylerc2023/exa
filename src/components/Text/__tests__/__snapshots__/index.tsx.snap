// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders body1 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 16,
      "fontWeight": "300",
      "letterSpacing": 0.5,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders body2 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 14,
      "fontWeight": "300",
      "letterSpacing": 0.25,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders button 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 14,
      "fontWeight": "600",
      "letterSpacing": 0.75,
      "opacity": 1,
    }
  }
>
  HELLO
</Text>
`;

exports[`renders button empty 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 14,
      "fontWeight": "600",
      "letterSpacing": 0.75,
      "opacity": 1,
    }
  }
>
  
</Text>
`;

exports[`renders caption 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 12,
      "fontWeight": "300",
      "letterSpacing": 0.4,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 14,
      "fontWeight": "300",
      "letterSpacing": 0.25,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders empty 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 14,
      "fontWeight": "300",
      "letterSpacing": 0.25,
      "opacity": 1,
    }
  }
>
  
</Text>
`;

exports[`renders h1 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 64,
      "fontWeight": "100",
      "letterSpacing": -1.5,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders h2 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 40,
      "fontWeight": "100",
      "letterSpacing": -0.5,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders h3 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 32,
      "fontWeight": "300",
      "letterSpacing": 0,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders h4 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 22.666666666666668,
      "fontWeight": "300",
      "letterSpacing": 0.25,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders h5 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 16,
      "fontWeight": "300",
      "letterSpacing": 0,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders h6 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 13.333333333333334,
      "fontWeight": "600",
      "letterSpacing": 0.15,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders overline 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 10,
      "fontWeight": "300",
      "letterSpacing": 1.5,
      "opacity": 1,
    }
  }
>
  HELLO
</Text>
`;

exports[`renders subtitle1 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 16,
      "fontWeight": "300",
      "letterSpacing": 0.15,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;

exports[`renders subtitle2 1`] = `
<Text
  style={
    Object {
      "color": "hsla(0, 0%, 0%, 1)",
      "fontSize": 14,
      "fontWeight": "600",
      "letterSpacing": 0.1,
      "opacity": 1,
    }
  }
>
  hello
</Text>
`;
